hearth/lib/hearth/middleware/send.rb

```diff
@@ -27,27 +27,29 @@ def initialize(_app, client:, stub_responses:,
       # @param input
       # @param context
       # @return [Output]
+      # rubocop:disable Metrics/MethodLength
       def call(input, context)
+        output = Output.new
         if @stub_responses
           stub = @stubs.next(context.operation_name)
-          output = Output.new
           apply_stub(stub, input, context, output)
           if context.response.body.respond_to?(:rewind)
             context.response.body.rewind
           end
-          output
         else
-          @client.transmit(
+          resp_or_error = @client.transmit(
             request: context.request,
             response: context.response
           )
-          Output.new
+          if resp_or_error.is_a?(Hearth::NetworkingError)
```

Comment:
An alternative is rescuing here

Reply:
Yeah I'd prefer rescuing here and having the http client throw the error rather than return either and error or response.